sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/samplers/ImmutableSamplingResult.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.samplers;
+
+import static io.opentelemetry.api.common.AttributeKey.doubleKey;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+@AutoValue
+abstract class ImmutableSamplingResult implements SamplingResult {
```

Comment:
I think we have a pattern of naming an AutoValue class `ImmutableBlah` so used that but not sure how cosistent are on it currently. It's a private class anyways